### PR TITLE
Incorporate Sterling's release notes feedback

### PR DESCRIPTION
**Pull request by nathan-contino**
- removed trailing comma from contributors
- trimmed wide snippets that didn't display nicely
- trimmed some unneeded words
- reordered to better match preamble

TODO: add link to DSL method [here](https://docs.gradle.org/nightly/userguide/more_about_tasks.html#sec:using_a_predicate) 

### Context
[Rendered notes preview](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/56681404:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220930171531%2B0000/docs/release-notes.html)

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by nathan-contino** [comment]
@tresat Sterling raised an interesting point -- the release notes "promoted" section currently contains lots of deprecated methods. Do we need to restructure it?

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Review by hythloda — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc [hunk 1]
```diff
@@ -365,7 +365,7 @@ Gradle offers multiple ways to skip the execution of a task.
 [[sec:using_a_predicate]]
 === Using a predicate
 
-You can use the `onlyIf()` method to attach a predicate to a task. The task's actions are only executed if the predicate evaluates to true. You implement the predicate as a closure. The closure is passed the task as a parameter, and should return true if the task should execute and false if the task should be skipped. The predicate is evaluated just before the task is due to be executed.
+You can use link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:onlyIf(groovy.lang.Closure)[`Task.onlyIf`] to attach a predicate to a task. The task's actions are only executed if the predicate evaluates to true. You implement the predicate as a closure. The closure is passed the task as a parameter, and should return true if the task should execute and false if the task should be skipped. The predicate is evaluated just before the task is due to be executed.
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:368]: Let's link to the Spec version instead.

```suggestion
You can use link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:onlyIf(org.gradle.api.specs.Spec)[`Task.onlyIf`] to attach a predicate to a task. The task's actions are only executed if the predicate evaluates to true. The predicate is passed the task as a parameter, and should return true if the task should execute and false if the task should be skipped. The predicate is evaluated just before the task is executed.
```

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -456,24 +457,12 @@ ADD RELEASE FEATURES ABOVE
...
-Any usages of these elements should be replaced by the now stable `getTestSources()` and `getTestResources()` methods and their respective setters.
-These new methods return and are backed by `ConfigurableFileCollection` instances for improved flexibility in how these collections of files can be used.
-Gradle now warns upon usage of these deprecated methods that they will be removed in Gradle 8.0.
-
-### Replacement methods in `org.gradle.api.tasks.testing.TestReport`
-
-The `getDestinationDir()`, `setDestinationDir(File)`, and `getTestResultsDirs()` and `setTestResultsDirs(Iterable)` methods are now `@Deprecated`.
-Any usages of them should be replaced by the now stable `getDestinationDirectory()` and `getTestResults()` methods and their associated setters.
-These deprecated elements will be removed in Gradle 8.0.
-
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md:476]: We still need part of this.

`getTestSources` and `getTestResources` are stable/promoted now.
`getDestinationDirectory` and `getTestResults` are stable/promoted now.

**Review by big-guy — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (hythloda) [reply subprojects/docs/src/docs/release/notes.md:476]: I like the old usage with the new but as long as this is in the migration guide I am okay with removing.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 3]
```diff
@@ -155,6 +155,19 @@ These members will be removed in Gradle 9.0.
 ==== Internal DependencyFactory was renamed
 The internal `org.gradle.api.internal.artifacts.dsl.dependencies.DependencyFactory` type was renamed to `org.gradle.api.internal.artifacts.dsl.dependencies.DependencyFactoryInternal`. As an internal type, it should not be used, but for compatibility reasons the inner `ClassPathNotation` type is still available. This name for the type is deprecated and will be removed in Gradle 8.0. The public API for this is on `DependencyHandler`, with methods such as `localGroovy()` providing the same functionality.
 
+==== Replacement collections in `org.gradle.plugins.ide.idea.model.IdeaModule`
+
+The `testResourcesDirs` and `testSourcesDirs` fields and their getters and setters have been deprecated.
+Replace usages with the now stable `getTestSources()` and `getTestResources()` methods and their respective setters.
+These new methods return and are backed by `ConfigurableFileCollection` instances for improved flexibility of use.
+Gradle now warns upon usage of these deprecated methods. They will be removed in Gradle 8.0.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:163]:
```suggestion
Gradle now warns upon usage of these deprecated methods. They will be removed in a future version of Gradle.
```

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (nathan-contino) [reply subprojects/docs/src/docs/release/notes.md:476]: Added these back to promotions.

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Pull request merged**